File: backend/modules/workout/src/main/java/com/micoach/workout/domain/Muscle.java
package com.micoach.workout.domain;

import lombok.Getter;

/**
 * Catálogo de músculos agrupados por zona corporal (tabla workout_muscles).
 */
@Getter
public class Muscle {

    private final Long id;
    private final String code;
    private final String name;
    private final String muscleGroup;

    private Muscle(Long id, String code, String name, String muscleGroup) {
        this.id = id;
        this.code = code;
        this.name = name;
        this.muscleGroup = muscleGroup;
    }

    public static Muscle restore(Long id, String code, String name, String muscleGroup) {
        return new Muscle(id, code, name, muscleGroup);
    }
}

File: mobile/lib/features/workout/domain/muscle.dart
// =====================================================================
// MiCoach — Catálogo de músculos.
// =====================================================================
class Muscle {
  final int id;
  final String code;
  final String name;
  final String muscleGroup;

  const Muscle({required this.id, required this.code, required this.name, required this.muscleGroup});

  factory Muscle.fromJson(Map<String, dynamic> json) => Muscle(
        id: json['id'] as int,
        code: json['code'] as String,
        name: json['name'] as String,
        muscleGroup: json['muscleGroup'] as String,
      );
}
